src/black/nodes.py

```diff
@@ -821,7 +821,9 @@ def is_type_comment(leaf: Leaf, suffix: str = "") -> bool:
     Only returns true for type comments for now."""
     t = leaf.type
     v = leaf.value
-    return t in {token.COMMENT, STANDALONE_COMMENT} and v.startswith("# type:" + suffix)
+    return t in {token.COMMENT, STANDALONE_COMMENT} and (
+        v.startswith(f"# pyright:{suffix}") or v.startswith(f"# type:{suffix}")
```

Comment:
```suggestion
        v.startswith((f"# pyright:{suffix}", f"# type:{suffix}"))
```
Is marginally more performant probably